specification/sbi_atee.adoc

```diff
@@ -10,8 +39,1296 @@ shown in the figure above, the TEE Runtime (operating in VS-mode) supports
+configure a boot VCPU by adding a `tvm_vcpu_id` with a value that specified for `tvm_boot_vcpuid`
+in the `tvm_create_params` structure that was used with sbi_tee_tvm_create().
+
+The possible error codes returned in `sbiret.error` are shown below.
+
+[#table_sbi_tee_host_create_tvm_vcpu_errors]
+.TEE Host Create TVM VCPU Errors
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code            | Description
+| SBI_SUCCESS           | The operation completed successfully.
+| SBI_ERR_INVALID_PARAM | `tvm_guest_id` or `tvm_vcpu_id` were invalid,
+                          or the TVM wasn't in `TVM_INITIALIZING` state.
+| SBI_ERR_FAILED        | The operation failed for unknown reasons.
+|===
+
+=== Function: TEE Host Run TVM VCPU (FID #16)
+[source, C]
+-----
+struct sbiret sbi_tee_host_run_tvm_vcpu(unsigned long tvm_guest_id,
+                                        unsigned long tvm_vcpu_id);
+-----
+Runs the VCPU specified by `tvm_vcpu_id` in the TVM specified by `tvm_guest_id`.
+The `tvm_guest_id` must be in a "runnable" state (requires a prior call 
+to `sbi_tee_host_finalize_tvm()`). The function does not return unless the TVM exits with
+a trap that cannot be handled by the TSM.
+
+*Returns* 0 on success in sbiret.value if the TVM exited with a resumable VCPU interrupt or exception,
+and non-zero otherwise. In the latter case, attempts to call `sbi_tee_host_run_tvm_vcpu()` with the
+same `tvm_vcpu_id` will fail.
+
+The possible error codes returned in `sbiret.error` are shown below.
+
+[#table_sbi_tee_host_run_tvm_vcpu_errors]
+.TEE Host Run TVM VCPU Errors
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code            | Description
+| SBI_ERR_SUCCESS       | The TVM exited, and sbiret.value contains 0 if the
+                        | interrupt or exception is resumable. The host can
+                        | examine `scause` to determine details.
+| SBI_ERR_INVALID_PARAM | `tvm_guest_id` or `tvm_vcpu_id` were invalid, or the
+                          TVM wasn't in `TVM_RUNNABLE` state.
+| SBI_ERR_FAILED        | The operation failed for unknown reasons.
+|===
+
+The TSM updates the `scause` field in the `tvm_vcpu_supervisor_csrs` region in the
+shared-memory for the VCPU that was previously configured by the host. The host should
+use the `scause` field to determine whether the exit was caused by an interrupt or exception,
+and then use the additional information to in the shared-memory region to determine further
+course of action (if sbiret.value is 0).
+
+The TSM sets the most significant bit in `scause` to indicate that that the exit was caused
+by an interrupt, and if this bit is clear, the implication is that the the exit was caused
+by an exception. The remaining bits specific information about the interrupt or exception,
+and the specific reason can be determined using the enumeration detailed below.
+
+[source, C]
+-------
+enum tvm_interrupt_exit {
```

Comment:
Can we just point to the priv spec instead of defining an enum that matches exactly the constants defined in the priv spec?